packages/mermaid/src/diagrams/xychart/chartBuilder/components/axis/baseAxis.ts

```diff
@@ -0,0 +1,421 @@
+            fill: this.axisThemeConfig.titleColor,
+            fontSize: this.axisConfig.titleFontSize,
+            rotation: 270,
+            verticalPos: 'top',
+            horizontalPos: 'center',
+          },
+        ],
+      });
+    }
+    return drawableElement;
+  }
+  private getDrawaableElementsForBottomAxis(): DrawableElem[] {
+    const drawableElement: DrawableElem[] = [];
+    if (this.showAxisLine) {
+      const y = this.boundingRect.y + this.axisConfig.axisLineWidth / 2;
+      drawableElement.push({
+        type: 'path',
+        groupTexts: ['bottom-axis', 'axis-line'],
+        data: [
+          {
+            path: `M ${this.boundingRect.x},${y} L ${
+              this.boundingRect.x + this.boundingRect.width
+            },${y}`,
+            strokeFill: this.axisThemeConfig.axisLineColor,
+            strokeWidth: this.axisConfig.axisLineWidth,
+          },
+        ],
+      });
+    }
+    if (this.showLabel) {
+      drawableElement.push({
+        type: 'text',
+        groupTexts: ['bottom-axis', 'label'],
+        data: this.getTickValues().map((tick) => ({
+          text: tick.toString(),
+          x: this.getScaleValue(tick),
+          y:
+            this.boundingRect.y +
+            this.axisConfig.labelPadding +
+            (this.showTick ? this.axisConfig.tickLength : 0) +
+            (this.showAxisLine ? this.axisConfig.axisLineWidth : 0),
+          fill: this.axisThemeConfig.labelColor,
+          fontSize: this.axisConfig.labelFontSize,
+          rotation: 0,
+          verticalPos: 'top',
+          horizontalPos: 'center',
+        })),
+      });
+    }
+    if (this.showTick) {
+      const y = this.boundingRect.y + (this.showAxisLine ? this.axisConfig.axisLineWidth : 0);
+      drawableElement.push({
+        type: 'path',
+        groupTexts: ['bottom-axis', 'ticks'],
+        data: this.getTickValues().map((tick) => ({
+          path: `M ${this.getScaleValue(tick)},${y} L ${this.getScaleValue(tick)},${
+            y + this.axisConfig.tickLength
+          }`,
+          strokeFill: this.axisThemeConfig.tickColor,
+          strokeWidth: this.axisConfig.tickWidth,
+        })),
+      });
+    }
+    if (this.showTitle) {
+      drawableElement.push({
+        type: 'text',
+        groupTexts: ['bottom-axis', 'title'],
+        data: [
+          {
+            text: this.title,
+            x: this.range[0] + (this.range[1] - this.range[0]) / 2,
+            y:
+              this.boundingRect.y +
+              this.boundingRect.height -
+              this.axisConfig.titlePadding -
+              this.titleTextHeight,
+            fill: this.axisThemeConfig.titleColor,
+            fontSize: this.axisConfig.titleFontSize,
+            rotation: 0,
+            verticalPos: 'top',
+            horizontalPos: 'center',
+          },
+        ],
+      });
+    }
+    return drawableElement;
+  }
+  private getDrawaableElementsForTopAxis(): DrawableElem[] {
```

Comment:
May be `renderable` or something else would be better, because `drawable` could refer to a drawer, but that is not the point:
```suggestion
  private getDrawableElementsForTopAxis(): DrawableElem[] {
```

Reply:
Changing the spelling mistake, and changing the name would be a very tedious task.

Reply:
Fixing mistake is enough